hoomd/mpcd/init.py

```diff
@@ -71,6 +71,67 @@ def make_random(N, kT, seed):
     hoomd.context.current.mpcd = data.system(_mpcd.SystemData(sysdef,pdata))
     return hoomd.context.current.mpcd
 
+def make_random_sphere(density, R, kT, seed):
+    R"""Initialize particles randomly inside a sphere
+
+    Args:
+        density (float): density of MPCD particles inside sphere
+        R (float): Radius of Sphere 
+        kT (float): Temperature of MPCD particles (in energy units)
+        seed (int): Random seed for initialization
+
+    Returns:
+        Initialized MPCD system data inside a sphere (:py:class:`hoomd.mpcd.data.system`)
+
+    MPCD particles are randomly initialized inside the sphere.
+    An MPCD system can be randomly initialized only **after** the HOOMD system
+    is first initialized (see :py:mod:`hoomd.init`). The system can only be
+    initialized one time. Random particle positions are drawn uniformly in
+    the sphere of radius *R* using rejection sampling within the box. Hence,
+    the number of particles initialized may not be equal to :math:`4 \pi R^3 \rho/3`,
+    where :math: `\rho` is the *density*. Particle velocities are drawn from a
+    Maxwell-Boltzmann distribution consistent with temperature *kT*. All MPCD
+    particles are given unit mass and type A.
+
+    Examples::
+
+        mpcd.init.make_random_sphere(density=5.0, R=50., kT=1.0, seed=42)
+
+    Notes:
+        Random number generation is performed using C++11 ``mt19937`` seeded by
+        *seed* plus the rank number in MPI simulations. This random number
+        generator is separate from other generators used in MPCD, so *seed* can
+        be reasonably recycled elsewhere.
+
+    """
+    hoomd.util.print_status_line()
+
+    if not hoomd.init.is_initialized():
+        hoomd.context.msg.error("mpcd: HOOMD system must be initialized before mpcd\n")
+        raise RuntimeError("HOOMD system not initialized")
+
+    if hoomd.context.current.mpcd is not None:
+        hoomd.context.msg.error("mpcd: system is already initialized, cannot reinitialize\n")
+        raise RuntimeError("mpcd system already initialized")
+
+    # make particle data first
+    sysdef = hoomd.context.current.system_definition
+    box = sysdef.getParticleData().getBox()
+    global_box = sysdef.getParticleData().getGlobalBox()
+    global_box_length = global_box.getL()
+    min_box_l = min([global_box_length.x, global_box_length.y, global_box_length.z])
+    if 2.*R > min_box_l :
```

Comment:
Since you are validating R here, you should also do the positivity check like you have in the C++ code:
```suggestion
    if R <= 0:
        # raise error like you have in C++ code
    elif 2.*R > min_box_l :
```